scripts/db/src/login.ts

```diff
@@ -1,113 +1,49 @@
-     */
-    const {
-      MONGO_URI,
-      MONGO_DB_NAME,
-    } = (configuration as unknown) as DbConfiguration
-
-    /**
-     * Logic in this function tries to match:
-     * EAForum/packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx
-     */
-    async function loginForumUser(): Promise<CallbackUser | null> {
-      // Connect to the Forum DB
-      const mongoClient: MongoClient = new MongoClient(MONGO_URI)
-      await mongoClient.connect()
-
-      // Query the users collection for someone with our email
-      const matchingUsers = await mongoClient
-        .db(MONGO_DB_NAME)
-        .collection<ForumUser>('users')
-        .find({ 'emails.address': email })
-        .collation({ locale: 'en', strength: 2 })
-        .toArray()
-
-      await mongoClient.close()
-
-      if (!matchingUsers.length) {
-        return null
-      }
-      if (matchingUsers.length > 1) {
-        throw new Error('More than one user with this email address')
-      }
-      const forumUser = matchingUsers[0]
-
-      // Check their password
-      // Meteor hashed its passwords twice, once on the client and once again on
-      // the server. To preserve backwards compatibility with Meteor passwords,
-      // we do the same, but do it both on the server-side
-      const meteorClientSideHash = createHash('sha256')
-        .update(password)
-        .digest('hex')
-      const isValid = await bcrypt.compare(
-        meteorClientSideHash,
-        forumUser.services.password.bcrypt
-      )
-      if (!isValid) {
-        return null
-      }
-
-      // Just in case we were dumb and did a stupid manual operation in our
-      // overly-flexible database
-      if (!Array.isArray(forumUser.emails)) {
-        throw new Error('Expected emails field to be an array')
-      }
-
-      // Which email did we find?
-      const emailInfo = forumUser.emails.find(
-        (e) => e.address.toLowerCase() === email.toLowerCase()
-      )
-      if (!emailInfo) {
-        // This should never happen, as they were returned by mongo because that
-        // field matched
-        throw new Error(
-          `User found by email ${email}, does not have that email`
-        )
-      }
-
-      return {
-        id: forumUser._id,
-        nickname: forumUser.displayName,
-        email: emailInfo.address,
-        email_verified: emailInfo.verified,
-      }
-    }
-
-    /** Give priority to Forum users, as the integration is newer and it has more users */
-    const forumUser = await loginForumUser()
-    if (forumUser) {
-      return callback(null, forumUser)
-    }
-    return callback(new WrongUsernameOrPasswordError(email))
-  } catch (err) {
-    return callback(err as Error)
-  }
+import { DbScriptCallback } from '../types/db-types'
+
+function login(email: string, password: string, callback: DbScriptCallback) {
+  // This script should authenticate a user against the credentials stored in
```

Comment:
I'm a bit confused by this. I imagine the point is that we don't store credentials in the database, so if auth0 reaches this point it means that `The user's credentials are invalid`.

If so we could add it as a comment?

```suggestion
function login(email: string, password: string, callback: DbScriptCallback) {
  // We don't store credentials in the database, so if auth0 reached this point
  // it means the user's credentials are invalid. Below is the original documentation


  // This script should authenticate a user against the credentials stored in
```

I'm surprised that there is no way to disable this completely in auth0